### DISCUSSION TITLE
Render error page from module

### DISCUSSION THREAD

**Discussion by siddharth223437** [nuxt/nuxt#17426] [Questions]
I want to render error component from my nuxt module.  Is there any way that I can use in my module ?

**Comment by Flanker72** [comment]
Just in theory: seems you can use error hooks to achieve the desired result: https://v3.nuxtjs.org/guide/features/error-handling
In your module you can register plugin/handler and override the default behaviour.

Another option is to override default error handler using routes modification (ex.: for 404 pages).

**Comment by alphaCastor** [comment]
@siddharth223437 I think you can achieve it by adding components inside you module src.

Your module.ts should look like something like this:

```js
import { defineNuxtModule } from '@nuxt/kit'

export default defineNuxtModule({
   // your code here
   hooks: {
      'components:dirs'(dirs) {
        // Add ./components dir to the list
        dirs.push({
          path: path.resolve(__dirname, './components'),
        })
    },
   }
})
```

After that if you will use your module inside app you will be able to use its components without importing them

UPD. If you want to use built module. You have to put your components inside runtime folder
